tools/convert-old-video-settings-file.py

```diff
@@ -0,0 +1,199 @@
+    Move old files, such as vidformat and gstreamer2 parameter files
+    to the user trash folder.
+"""
+
+import os
+import subprocess
+import sys
+
+
+def merge_gst_elements(elements: list):
+    """A simple helper to merge gstreamer elements in pipeline format
+
+    Args:
+        elements (list): List that contains gstreamer elements
+
+    Returns:
+        str: Final gstreamer pipeline
+    """
+
+    return ' ! '.join(elements)
+
+
+class Convert:
+    """Convert main class
+        Check file description for more information
+    """
+
+    HOME = os.path.expanduser('/tmp')
+    TRASH = os.path.join(HOME, '.local/share/Trash/files')
+    vidformat_file = os.path.join(HOME, 'vidformat.param')
+    gstreamer2_file = os.path.join(HOME, 'gstreamer2.param')
+
+    device_settings = {
+        'device': '/dev/video0',
+        'width': 1920,
+        'height': 1080,
+        'framerate': 30,
+    }
+
+    SOURCE_PIPELINE = merge_gst_elements([
+        'v4l2src device={device} do-timestamp=true',
+        'video/x-h264, width={width}, height={height}, framerate={framerate}/1',
+    ])
+
+    PARSE_PIPELINE = merge_gst_elements([
+        'h264parse',
+        'queue',
+        'rtph264pay config-interval=10 pt=96',
+        'udpsink host=192.168.2.1 port=5600',
+    ])
+
+    def update_configuration(self, pair: tuple):
+        """Update class internal configuration about video source and resolution
+
+        Args:
+            pair (tuple): Tuple that contains (key, value) for pipeline
+            settings dict
+        """
+
+        (key, value) = pair
+        self.device_settings[key] = value
+
+    @staticmethod
+    def create_source_element(device='/dev/video0', width=1920, height=1080, framerate=30):
+        """Populate and generate gstreamer v4l2src pipeline
+
+        Args:
+            device (str, optional): Video device. Defaults to '/dev/video0'.
+            width (int, optional): Video width. Defaults to 1920.
+            height (int, optional): Video height. Defaults to 1080.
+            framerate (int, optional): Video framerate. Defaults to 30.
+
+        Returns:
+            str: Return string with the pipeline
+        """
+
+        return Convert.SOURCE_PIPELINE.format(
+            device=device,
+            width=width,
+            height=height,
+            framerate=framerate
+        )
+
+    @staticmethod
+    def read_content_then_trash_file(filepath: str):
+        """Get content from file and move it to the trash
+
+        Args:
+            filepath (str): Path for the parameter file
+
+        Returns:
+            str: File content
+        """
+
+        if not os.path.isfile(filepath):
+            return None
+
+        file_object = open(filepath, 'r')
+        content = file_object.read()
+        file_object.close()
+
+        filename = os.path.basename(filepath)
+        # TODO: Remove this before merge
+        #os.rename(filepath, os.path.join(Convert.TRASH, filename))
```

Comment:
does this need to be uncommented?

I think that it would be better for the caller to move the old file to trash, in case there are errors after this is called

Reply:
Yes, I'll remove the comments and check any TODO that  was used for development.

Reply:
Done.